modules/platforms/dotnet/Apache.Ignite.Core/Impl/Client/Services/ServicesClient.cs

```diff
@@ -202,7 +217,127 @@ public IServicesClient WithServerKeepBinary()
+            {
+                if (Interlocked.CompareExchange(ref _updateInProgress, 1, 0) == 1)
+                    return;
+
+                var socket = _svcClient._ignite.Socket;
+
+                var topVer = socket.GetTopologyVersion();
+
+                var log = _svcClient._ignite.GetConfiguration().Logger;
+
+                var groupNodes = _svcClient._clusterGroup?.GetNodes();
+                
+                var top = await socket.DoOutInOpAsync(ClientOp.ServiceGetTopology,
+                    ctx => ctx.Writer.WriteString(_svcName),
+                    ctx =>
+                    {
+                        var cnt = ctx.Reader.ReadInt();
+
+                        var res = new List<Guid>(cnt);
+                        
+                        for (var i = 0; i < cnt; ++i)
+                            res.Add(BinaryUtils.ReadGuid(ctx.Reader.Stream));
+
+                        return res;
+                    }, (status, err) =>
+                    {
+                        log.Error("Failed to update topology of the service '" + _svcName + "'.", err);
+
+                        return _nodes;
+                    }).ConfigureAwait(false);
+                
+                _nodes = FilterTopology(top, groupNodes?.Select(n => n.Id).ToList());
+                
+                Interlocked.Exchange(ref _lastUpdateRequestTime, DateTime.Now.Ticks);
+                Interlocked.Exchange(ref _lastAffTop, topVer);
+
+                log.Debug("Topology of service '" + _svcName + "' has been updated. The " +
+                          "service instance nodes: " + string.Join(", ", top.Select(gid=>gid.ToString())) +
+                          ". Effective topology with the cluster group is: " + 
+                          string.Join(", ", _nodes.Select(gid=>gid.ToString())) + '.');
+
+                _updateInProgress = 0;
+            }
+
+            /// <summary>
+            /// Filters service topology regarding to the cluster group.
+            /// </summary>
+            private static IList<Guid> FilterTopology(IList<Guid> serviceTopology, IList<Guid> clusterGroup)
+            {
+                return clusterGroup == null ? serviceTopology : serviceTopology.Intersect(clusterGroup).ToList();
+            }
+
+            /// <summary>
+            /// Provides last known service topology and asynchronously updates it if required.
+            /// </summary>
+            internal IList<Guid> GetAndUpdate()
+            {
+                if (_updateInProgress != 0)
+                    return _nodes;
+
+                var curAff = _svcClient._ignite.Socket.GetTopologyVersion();
+                var lastKnownAff = Interlocked.Read(ref _lastAffTop);
+                var sinceLastUpdate = DateTime.Now.Ticks - Interlocked.Read(ref _lastUpdateRequestTime);
+
+                if (curAff > lastKnownAff || sinceLastUpdate > SrvTopUpdatePeriod)
+                    UpdateTopologyAsync().ConfigureAwait(false);
```

Comment:
```suggestion
                    _ = UpdateTopologyAsync().ConfigureAwait(false);
```

Discard the task explicitly to show that we did not skip `await` by mistake.